python/ctsm/args_utils.py

```diff
@@ -44,10 +44,10 @@ def plon_type(plon):
     Returns:
         plon_out (float): converted longitude between 0 and 360
     """
-    plon = float(plon)
-    if plon < -180 or plon > 360:
+    plon_float = float(plon)
```

Comment:
First, changing variable name from `plon` to `plon_float` is not a big deal and I can change this easily...

But I don't understand the motivation behind changing it and I want to clarify why... 

1) Python variable re-binding is allowed and in some cases saves memory.
For example:
```
a= '5'
a = int(5)
```
is allowed in Python because it is a dynamic programming language. 
Please see: https://stackoverflow.com/questions/48224911/is-it-good-practice-to-change-the-type-of-a-variable-in-python

In fact this (dynamic typing) is one of the features of Python, so why not use it?
https://www.pythontutorial.net/advanced-python/dynamic-typing-in-python/

2) When the type of a variable changes it is recommended to use another name only when the old variable (with old type) is being used in the code again, which is not the case here...

3) Now, let's look at this function closely. It is a function that is only used in the `argparser` type:
``` python
        pt_parser.add_argument('--lon',
                    help='Single point longitude. [default: %(default)s]', 
                    action="store",
                    dest="plon",
                    required=False,
                    **type = plon_type,**
                    default= 287.8 )
```
So the parser return to the user has a variable called `plon` (i.e. destination variable `des=plon`) with type float. So the user does not see/interact with string form of plon at all.... This is not a function that the user calls to change the variable for them rather it is a small helper function called by the argparse (with the method recommended by Python argparse). 

4. Finally, in Python it is always better to avoid using types in names to improve code readability... (hence the introduction of type hinting in python v3.6)... 

* It is advised in Python to avoid using the types in variable names because it is potentially dangerous and reduces readability. 
https://docs.quantifiedcode.com/python-anti-patterns/readability/putting_type_information_in_a_variable_name.html

* On the other hand, if there are two different variables with two types needed in the code by all means, it would have made sense to create `plon_str` or `plon_float` or whatever names...

Please check the accepted response here:
https://stackoverflow.com/questions/25489118/same-variable-name-for-different-types-in-python

* Using type hinting,  this is also 100% allowed :
```python
def just_test(var: int) -> str:
    var = 'Printing: '+ str(var)
    print (var)
    return var 
 
# first var is integer:
var = 2 
var = just_test(var)  
```